### PR TITLE
Replace controls windows and menus

### PR DESCRIPTION
**Pull request by nallath**
CURA-8683 

https://github.com/Ultimaker/Cura/pull/11236

### CODE REVIEW

File: UM/Qt/qml/UM/MenuItem.qml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        onActivated: root.triggered()
+        sequence: root.action!= null ? root.action.shortcut: null
+    }
+
+    function replaceText(txt)
+    {
+        var index = txt.indexOf("&")
+        if(index >= 0)
+        {
+            txt = txt.replace(txt.substr(index, 2), ("<u>" + txt.substr(index + 1, 1) +"</u>"))
```
Comment (casperlamboo) [inline UM/Qt/qml/UM/MenuItem.qml:25]:
```suggestion
            txt = txt.replace(txt.substr(index, 2), "<u>" + txt.substr(index + 1, 1) + "</u>")
```

File: UM/Qt/qml/UM/MenuItem.qml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    property alias shortcut: _shortcut.sequence
+    property bool indicatorVisible: root.icon.source.length > 0 || root.checkable
+
+    Shortcut
+    {
+        id: _shortcut
+        enabled: root.enabled
+        onActivated: root.triggered()
+        sequence: root.action!= null ? root.action.shortcut: null
```
Comment (casperlamboo) [inline UM/Qt/qml/UM/MenuItem.qml:17]:
```suggestion
        sequence: root.action != null ? root.action.shortcut: null
```

File: UM/Qt/qml/UM/MenuItem.qml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    contentItem: RowLayout
+    {
+        spacing: 0
+        width: root.width
+        opacity: root.enabled ? 1 : 0.5
+
+        Item
+        {
+            // Spacer
+            width: root.indicatorVisible ? root.indicator.width + 4 : 0
```
Comment (casperlamboo) [inline UM/Qt/qml/UM/MenuItem.qml:39]: Shouldn't we pick margin size from the theme.json here?

File: UM/Qt/qml/UM/Menu.qml [hunk 4]
```diff
@@ -9,10 +9,14 @@ Menu
...
     property bool shouldBeVisible: true
+    onParentChanged:  handleVisibility()
     function handleVisibility()
     {
-        root.parent.visible = shouldBeVisible
-        root.parent.height = shouldBeVisible ? UM.Theme.getSize("setting_control").height + UM.Theme.getSize("narrow_margin").height: 0
+        if(parent)
+        {
+            root.parent.visible = shouldBeVisible
+            root.parent.height = shouldBeVisible ? UM.Theme.getSize("setting_control").height + UM.Theme.getSize("narrow_margin").height: 0
```
Comment (casperlamboo) [inline UM/Qt/qml/UM/Menu.qml:19]: Aren't margins applied to both the top and bottom?

```suggestion
            root.parent.height = shouldBeVisible ? UM.Theme.getSize("setting_control").height + 2 * UM.Theme.getSize("narrow_margin").height : 0
```

File: UM/Qt/qml/UM/MenuItem.qml (hunk 3 above)
Comment (nallath) [reply UM/Qt/qml/UM/MenuItem.qml]: Whooops :$ You're right! Fixed in 66ebf83030a635e1e56bbcae936cfc9880c26a02

File: UM/Qt/qml/UM/MenuItem.qml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    contentItem: RowLayout
+    {
+        spacing: 0
+        width: root.width
+        opacity: root.enabled ? 1 : 0.5
+
+        Item
+        {
+            // Spacer
+            width: root.indicatorVisible ? root.indicator.width + UM.Theme.getSize("narrow_margin").width : 0
```
Comment (casperlamboo) [inline UM/Qt/qml/UM/MenuItem.qml:51]: Margins should be added to the left and right side of components
```suggestion
            width: root.indicatorVisible ? root.indicator.width + 2 * UM.Theme.getSize("narrow_margin").width : 0
```

**Comment by nallath** [comment]
Discussed with @casperlamboo that this could be merged after the last minor remarks were fixed.

**Pull request merged**